LibTest/async/MultiStreamController/onCancel_A02_t01.dart

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion FutureOr<void> Function()? onCancel
+/// The callback which is called when the stream is canceled.
+///
+/// May be set to `null`, in which case no callback will happen.
+///
+/// @description Checks that if this callback is set to `null`, then no callback
+/// happens.
+/// @author sgrekhov22@gmail.com
+
+import "dart:async";
+import "../../../Utils/expect.dart";
+
+main() {
+  asyncStart(2);
+  var stream = Stream<int>.multi((controller) {
+    controller.onCancel = () {
+      Expect.fail("Unexpected onCancel");
+    };
+    controller.onCancel = null;
```

Comment:
Yes, but... Is there any other way to check that "May be set to `null`, in which case no callback will happen."?

Reply:
I'd simply think that if `x` is a variable then `x = null;` would be an action which can be described as "`x` is set to null", and we wouldn't have to worry about the history.

In other words, we wouldn't have to try things like "setting `x` to null in a situation where it has the value `v`" for a bunch of different `v`, for instance the cases where `v` is `() { /*something*/ }` and where `v` is null.

The daunting part is that we basically cannot introduce this rule in general (that is, "check different histories when we're checking the effect of setting a variable to a specific value"), because it might very well make the complexity of a huge number of tests grow substantially. So that's the reason why I think we should be really careful about taking the history into account here.

That said, it is true that anything that looks like an instance variable in Dart could be a getter and a setter, and "reading/writing the variable" could give rise to arbitrary computations, and there might actually be a difference between the behavior of `x` in the situation where it has a specific value `v`, but different histories.

So you could argue that we should _always_ test with different histories when we're supposed to test that setting a variable to a specific value has a specific effect. I just don't think it's doable, or worthwhile. So let's only test that kind of thing when the history is mentioned in the documentation/specification of the feature being tested.